test/array.each.js

```diff
@@ -3,72 +3,183 @@
+    }
+  );
+});
+
+metatests.test('successful each / map', test => {
+  const map = new Map([[1, 'a'], [2, 'b'], [3, 'c']]);
+  const elementsArr = [];
 
   metasync.each(
-    arr,
+    map,
     (el, callback) =>
       process.nextTick(() => {
-        elementsSet.add(el);
+        elementsArr.push(el);
         callback(null);
       }),
     err => {
       test.error(err);
-      test.strictSame(elementsSet, expectedElementsSet);
+      test.strictSame(elementsArr, [...map]);
       test.end();
     }
   );
 });
 
-metatests.test('each with error', test => {
-  const arr = [1, 2, 3, 4];
-  let count = 0;
+metatests.test('successful each / string', test => {
+  const string = 'aaabcdeefff';
+  const elementsArr = [];
 
-  const elementsSet = new Set();
-  const expectedElementsCount = 2;
-  const eachError = new Error('Each error');
+  metasync.each(
+    string,
+    (el, callback) =>
+      process.nextTick(() => {
+        elementsArr.push(el);
+        callback(null);
+      }),
+    err => {
+      test.error(err);
+      test.strictSame(elementsArr, [...string]);
+      test.end();
+    }
+  );
+});
+
+metatests.test('each with empty / array', test => {
+  const arr = [];
+  const elementsArr = [];
 
   metasync.each(
     arr,
     (el, callback) =>
       process.nextTick(() => {
-        elementsSet.add(el);
-        count++;
-        if (count === expectedElementsCount) {
-          callback(eachError);
-        } else {
-          callback(null);
-        }
+        elementsArr.push(el);
+        callback(null);
+      }),
+    err => {
+      test.error(err);
+      test.strictSame(elementsArr, arr);
+      test.end();
+    }
+  );
+});
+
+metatests.test('each with empty / map', test => {
+  const map = new Map();
+  const elementsArr = [];
+
+  metasync.each(
+    map,
+    (el, callback) =>
+      process.nextTick(() => {
+        elementsArr.add(el);
+        callback(null);
       }),
     err => {
-      test.strictSame(err, eachError);
-      test.strictSame(elementsSet.size, expectedElementsCount);
+      test.error(err);
+      test.strictSame(elementsArr, [...map]);
       test.end();
     }
   );
 });
+
+metatests.test('each with empty / string', test => {
+  const string = '';
+  const elementsArr = [];
+
+  metasync.each(
+    string,
+    (el, callback) =>
+      process.nextTick(() => {
+        elementsArr.push(el);
+        callback(null);
+      }),
+    err => {
+      test.error(err);
+      test.strictSame(elementsArr, [...string]);
+      test.end();
+    }
+  );
+});
+
+metatests.test('each with empty / set', test => {
+  const set = new Set();
+
+  const elementsArr = [];
+
+  metasync.each(
+    set,
+    (el, callback) =>
+      process.nextTick(() => {
+        elementsArr.add(el);
+        callback(null);
+      }),
+    err => {
+      test.error(err);
+      test.strictSame(elementsArr, [...set]);
+      test.end();
+    }
+  );
+});
+
+// metatests.test('each with error', test => {
```

Comment:
Why this is commented out?

Reply:
Because now `each` works parallel and this callback `callback(eachError);` will not cause the stopping of processing other items.
What should I do with this test? rewrite for checking if other els were processed?